src/persistence/Elsa.Persistence.MongoDb/WorkflowEngineMongoDbClient.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+using Elsa.Data;
+using Elsa.Models;
+
+using Microsoft.Extensions.Options;
+
+using MongoDB.Driver;
+
+namespace Elsa.Persistence.MongoDb
+{
+    public class WorkflowEngineMongoDbClient
+    {
+        private readonly MongoClient _mongoClient;
+        private readonly IMongoDatabase _mongoDatabase;
+
+        public WorkflowEngineMongoDbClient(IOptions<ElsaMongoDbOptions> options)
+        {
+            _mongoClient = new MongoClient(options.Value.ConnectionString);
+            _mongoDatabase = _mongoClient.GetDatabase(options.Value.Db);
+   
+        }
+
+        public IMongoCollection<WorkflowDefinition> WorkflowDefinitions => _mongoDatabase.GetCollection<WorkflowDefinition>(CollectionNames.WorkflowDefinitions);
+        public IMongoCollection<WorkflowInstance> WorkflowInstances => _mongoDatabase.GetCollection<WorkflowInstance>(CollectionNames.WorkflowInstances);
+    }
```

Comment:
Instead of having this class, how about we just register `IMongoDatabase`, `IMongoCollection<WorkflowDefinition>` and `IMongoCollection<WorkflowInstance>` with DI directly, allowing any classes in the MongoDb persistence provider to inject the two Mongo collection types directly?

Reply:
Good idea, but I will do it a little differently. I need the WorkflowEngineMongoDbClient class for the client configuration.  I don't know how to configure the client otherwise in only one place.

`IMongoDatabase` should not be registered into the DI. If someone has already done it for his project, there will be conflicts.

I will register the collections like this:

`.AddScoped<IMongoCollection<WorkflowDefinition>>(services => services.GetRequiredService<ElsaMongoDbClient>().WorkflowDefinitions)`

`.AddScoped<IMongoCollection<WorkflowInstance>>(services => services.GetRequiredService<ElsaMongoDbClient>().WorkflowInstances)`